spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/Column.java

```diff
@@ -41,5 +43,31 @@
 	 * field which it annotates.
 	 * @return the name of the column in the Spanner table
 	 */
-	String name();
+	String name() default "";
+
+	/**
+	 * The maximum length of the column in Cloud Spanner terms. For example, for STRING
+	 * columns this refers to the number of characters. For BYTES columns this refers to
+	 * the number of bytes. This setting is only used when generating schema from an
+	 * entity class. A setting of 0 indicates an unlimited maximum length.
+	 *
+	 * @return the maximum length for the column
+	 */
+	long generateSchemaMaxLength() default 0;
```

Comment:
`spannerTypeMaxLength`?
Maybe `-1` to represent unlimited length, or use `null` with `Long`.

Reply:
Unfortunately `null` isn't considered a constant value and cannot be used in enum defaults. I can use -1

Reply:
OK, what about the attribute name suggestion?

Reply:
The `spannerType` ? I made that change...

Reply:
changing this one to `spannerTypeMaxLength`

Reply:
@meltsufin ah ok, also removed the "generateSchema". PTAL